### PR TITLE
Refresh frontend on start event

### PR DESCRIPTION
**Pull request by bdraco**
## Proposed change

Refresh the frontend when Home Assistant fires the start event.

This PR supports https://github.com/home-assistant/core/pull/36093#pullrequestreview-417473363

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes home-assistant/core#35924
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by bramkragten** [comment]
Although this probably fixes the issues with the early startup of the frontend, I don't think this is the way to go.

If startup took a long time we now just refresh the page when a user was maybe in the middle of editing something.

I rather fix the issues I mentioned in your PR, the first thing we will need for that is a flag so the frontend knows HA is not done starting yet.

**Comment by bdraco** [comment]
> I rather fix the issues I mentioned in your PR, the first thing we will need for that is a flag so the frontend knows HA is not done starting yet.

How should we tell the the frontend that HA isn't starting yet?

**Comment by bramkragten** [comment]
I'm completely fine with making the changes on this side :-)

We could add a started boolean to `hass.config`?

**Comment by bdraco** [comment]
> I'm completely fine with making the changes on this side :-)
TYVM 🥇 
> 
> We could add a started boolean to `hass.config`?

I added `hass.config.start_complete`. I didn't use `started` as I didn't want to confuse it with `EVENT_HOMEASSISTANT_STARTED`

File: src/state/connection-mixin.ts [hunk 1]
```diff
@@ -145,6 +145,11 @@ export const connectionMixin = <T extends Constructor<HassBaseEl>>(
         }
       });
 
+      // If they reconnect during startup we need to reload
+      // when startup is completed
+      conn.subscribeEvents(() => {
+        location.reload();
+      }, "homeassistant_start");
```
Comment (balloob) [inline src/state/connection-mixin.ts:152]: This is too aggressive. Most pages will actually do just fine. It's just generated Lovelace UIs opened while starting up that have issues.

Comment (balloob) [reply src/state/connection-mixin.ts:152]: We also shouldn't add non-data logic in the `state` mix-ins as they are used for other projects too (ie cast). We have logic like this in `home-assistant.ts` for when we detect a version upgrade (And a reload is needed)

Comment (bdraco) [reply src/state/connection-mixin.ts:152]: Agreed.  Probably makes sense to close this PR and go the route @bramkragten suggested above in https://github.com/home-assistant/frontend/pull/6042#issuecomment-633624026 

I'll leave this open and marked as a draft for now though so there is a place to comment.

**Comment by bdraco** [comment]
This work has been superseded by a better solution in https://github.com/home-assistant/frontend/pull/6068

**Pull request closed without merging**